core/templates/services/exploration-html-formatter.service.ts

```diff
@@ -78,94 +78,113 @@ export class ExplorationHtmlFormatterService {
       interactionCustomizationArgs: InteractionCustomizationArgs,
       parentHasLastAnswerProperty: boolean,
       labelForFocusTarget: string,
-      savedSolution: string | null): string {
-    var htmlInteractionId = this.camelCaseToHyphens.transform(interactionId);
-    var element = $('<oppia-interactive-' + htmlInteractionId + '>');
-
+      savedSolution: 'savedMemento()' | null
+  ): string {
+    let availableInteractionIds = Array.prototype.concat.apply(
+      [],
+      AppConstants.ALLOWED_INTERACTION_CATEGORIES.map(
+        category => category.interaction_ids
+      )
+    );
+    if (!availableInteractionIds.includes(interactionId)) {
+      throw new Error(`Invalid interaction id: ${interactionId}.`);
+    }
+    let htmlInteractionId = this.camelCaseToHyphens.transform(interactionId);
+    // The createElement is safe because we verify that the interactionId
+    // belongs to a list of interaction IDs.
+    let element = document.createElement(
+      `oppia-interactive-${htmlInteractionId}`);
     element = (
       this.extensionTagAssembler.formatCustomizationArgAttrs(
-        element, interactionCustomizationArgs));
-    const tagEnd = '></oppia-interactive-' + htmlInteractionId + '>';
-    let directiveOuterHtml = element.get(0).outerHTML.replace(tagEnd, '');
-    let spaceToBeAdded = true;
-    const getLastAnswer = (): string => {
-      let propValue = parentHasLastAnswerProperty ? 'lastAnswer' : 'null';
-      if (this.migratedInteractions.indexOf(interactionId) >= 0) {
-        return '[last-answer]="' + propValue + '"';
-      } else {
-        return 'last-answer="' + propValue + '"';
-      }
-    };
-    if (savedSolution) {
+        element, interactionCustomizationArgs)
+    );
+    // The setAttribute is safe because we verify that the savedSolution
+    // is 'savedMemento()'.
```

Comment:
Do we need an "else throw error" here?

Reply:
As I have verified that nothing else is passed there now and this is also checked by the TS typing I think it is fine to skip else throw here.